daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -48,9 +48,10 @@ final case class ScenarioRunner(
     var steps = 0
     while (!machine.isFinal) {
       //machine.print(steps)
-      machine.step match {
-        case SResultContinue =>
-          steps += 1
+      steps += 1 // this counts the number of external `Need` interactions
+      val res: SResult = machine.run()
+      res match {
+        case SResultFinalValue(_) =>
```

Comment:
I don't see a consequence for this match branch.

Reply:
This is legal scala syntax. It means the same as having `()` on RHS of the `=>`
It's used all over the scala code base.
But I agree, I generally prefer when code is explicit with `()`, but I missed it here.

Reply:
Since that code has changed anyway, can we please add the `()` here.

Reply:
ok